### PR TITLE
Firefox & Safari support

### LINKED ISSUE #108: Support Firefox & Safari for .encrypt & .saveZip
Conflux [uses a ponyfill](https://github.com/transcend-io/conflux/blob/5e53d10c7e44ab3db7d4b3ef8738ebc1e7436b85/src/write.js#L10) for Safari transform streams. Is this an applicable solution for Penumbra `.encrypt`?

### PR DESCRIPTION
**Pull request by eligrey**
This PR adds Firefox & Safari support for all Penumbra APIs. As a side effect of our new more efficient streaming architecture, browsers without native streaming support should now have a potentially higher max main thread file size.

This also fixes https://penumbra.netlify.app/ on Firefox & Safari, though it can be quite slow on the main thread.

## Related issues
 
- Closes #108

## Compatibility

✅ = Full support

🟡 = 32 MiB limit (potentially higher)

🟠 = 16 MiB limit

❌ = No support

❓ = Unknown

### Before

|          | .decrypt | .encrypt | .saveZip |
| -------- | -------: | -------: | -------: |
| Chrome   |       ✅ |       ✅ |       ✅ |
| Edge >18 |       ✅ |       ✅ |       ✅ |
| Safari   |       🟠 |       ❌ |       ❌ |
| Firefox  |       🟠 |       ❌ |       ❌ |
| Edge 18  |       ❌ |       ❌ |       ❌ |

### After

|          | .decrypt | .encrypt | .saveZip |
| -------- | -------: | -------: | -------: |
| Chrome   |       ✅ |       ✅ |       ✅ |
| Edge >18 |       ✅ |       ✅ |       ✅ |
| Safari   |       🟡 |       🟡 |       🟡 |
| Firefox  |       🟡 |       🟡 |       🟡 |
| Edge 18  |       ❌ |       ❌ |       ❌ |

### CODE REVIEW

File: src/zip.ts [hunk 1]
```diff
@@ -28,6 +31,12 @@ const sumWrites = async (writes: Promise<number>[]): Promise<number> => {
   );
 };
 
+// const { createWriteStream } = streamSaver;
```
Comment (michaelfarrell76) [inline src/zip.ts:34]: whats this?

**Review by michaelfarrell76 — APPROVED** [review]
this is huge!

**Comment by michaelfarrell76** [comment]
can we link any issues that this closes?

File: src/zip.ts (hunk 1 above)
Comment (eligrey) [reply src/zip.ts:34]: Remnants of a snippet that was moved to API.ts. I've just deleted this

**Pull request merged**